### PR TITLE
Suppress color conversion warning when converting invalid LAB coordinates

### LINKED ISSUE #5311: New Label Layer emit a warning of invalid color
Since #5180, now that we rely on skimage newer version, we actually see warning about improper colors and colorspace. 

In particular we get "Color data out of range: Z < 0 in N pixels"

The warning in napari emanates from this line:

https://github.com/napari/napari/blob/1cc90a74211c6cbf3fc644da5866f9752cf1ac78/napari/utils/colormaps/colormap_utils.py#L361

And it seem that we are sampling colors outside of the lab space. Skimage goes via the XYZ space, and Z is negative for some of those colors, and then clipped to 0, then converted from XYZ to RGB. 

I'm not sure what the implications are – there are at least some improvement we can send upstream to skimage to tell us _which_ values are incorrect. 

_Originally posted by @Carreau in https://github.com/napari/napari/issues/5180#issuecomment-1306910267_

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description
Our default `Labels` colormap is generated by sampling uniformly at random from the unit cube. These 3-tuples are intended to be coordinates in the CIELAB colorspace, but some of those coordinates may not be valid since that space is not linear. We then convert those coordinates to their corresponding RGB vectors. The scikit-image color conversion function handles this by projecting those invalid vectors, but it also issues a warning when doing so. This whole process approximates sampling uniformly at random over the CIELAB colorspace and the projection adds a little more error to the approximation, but overall it's good enough.

Originally (in #175), we vendored code from scikit-image to handle color space conversion and removed this warning in our copy (see https://github.com/napari/napari/commit/d1bb7152cb8949d23ad4e41cacab13211f9e4a27). Later, we brought in scikit-image as a dependency. Then recently (in #5180) we removed the vendored code, which caused the warning to be issued (#5311).

This PR specifically suppresses that warning.

It also expands on the explanatory comment describing how the min and max values in the LAB and LUV spaces were precomputed.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# References
Closes #5311 

# How has this been tested?
- [ ] all existing tests pass with my change

I tested this manually. I also wrote a simple test to reproduce the error, but it's probably not worth maintaining, so I did not add it to this PR.

### CODE REVIEW

**Comment by andy-sweet** [comment]
@Carreau : I would expect this warning to be issued in any test that creates a `Labels` layer with mostly default parameter values (of which there are many), and thus fail. Do we specifically ignore this warning in the test configuration somewhere? I couldn't find it.

**Comment by andy-sweet** [comment]
@jni : I also added you as a reviewer here since you made some of the original additions. Since this is just a warning, I don't think there's too much rush on this.

File: napari/utils/colormaps/colormap_utils.py [hunk 1]
```diff
@@ -358,7 +368,17 @@ def _color_random(n, *, colorspace='lab', tolerance=0.0, seed=0.5):
         elif colorspace == 'rgb':
             raw_rgb = random
         else:  # 'lab' by default
-            raw_rgb = colorconv.lab2rgb(random * LABRNG + LABMIN)
+            # The values in random are in [0, 1], but since the LAB colorspace
+            # is not exactly contained in the unit-box, some 3-tuples might not
+            # be valid LAB color coordinates. scikit-image handles this by projecting
+            # such coordinates into the colorspace, but will also warn when doing this.
+            with warnings.catch_warnings():
```
Comment (andy-sweet) [inline napari/utils/colormaps/colormap_utils.py:375]: I think the same problem (i.e. invalid coordinates) could occur for the LUV colorspace, but the corresponding scikit-image function does not seem to warn in that case, so I left it alone for now.

File: napari/utils/colormaps/colormap_utils.py [hunk 2]
```diff
@@ -118,12 +118,22 @@ def _all_rgb():
     return np.stack((r, g, b), axis=-1).reshape((-1, 3))
 
 
-# obtained with colorconv.rgb2luv(_all_rgb().reshape((-1, 256, 3)))
+# The following values were precomputed and stored as constants
```
Comment (kephale) [inline napari/utils/colormaps/colormap_utils.py:121]: Is it possible to do this at install time? Not that I think that is necessary in this case...

**Review by kephale — APPROVED** [review]
This looks good. My only comment is a curiosity, not a requested change.

File: napari/utils/colormaps/colormap_utils.py (hunk 2 above)
Comment (andy-sweet) [reply napari/utils/colormaps/colormap_utils.py:121]: I'm not aware of a way to do that in Python, but that doesn't mean it doesn't exist!

**Review by jni — APPROVED** [review]
Good stuff! Thank you @andy-sweet!

**Pull request merged**